### DISCUSSION TITLE
Looking for advice for a data model with ranges of units

### DISCUSSION THREAD

**Discussion by marcus-at-localhost** [atrocore/atrocore#578] [Q&A] [answered]
We have a product attribute called Temperature and it's a [unit field](https://help.atrocore.com/#/atropim/user-guide/attributes?id=available-attribute-types).

We generate a datasheet with the API and display the temperature like this:

- AttributeX: 1
- Temperature: 50.5 °C
- etc

There are only a few products that have to show a range of temperatures:

- AttributeX: 1
- Temperature: 40.3 - 50.5 °C
- etc

Product managers asked me to make the Unit field a simple Text input to write in the range, which makes it impossible to use filter on that field and I believe that's a bad data model.

Instead I created a field "min Temperature"

- AttributeX: 1
- min Temperature: 40.3 °C
- Temperature: 50.5 °C
- etc

Now this is straight forward in the data model, for the presentation layer, I have to transform these two specific fields, to print it like this: 

- Temperature: 40.3 - 50.5 °C

or transform "Temperature" to "max. Temperature"  if there is a "min Temperature" value, to list it like this:

- min Temperature: 40.3 °C
- max Temperature: 50.5 °C

I'd like to keep the data sheet template as unspecific as possible, but maybe I have to go in and add this edge case and transform the data or better the label somehow (it's multilingual, just slapping a max in front of it probably won't do it :'-).

This has not necessarily something to do with the PIM, but maybe someone came across an edge case like this and can give advice how to deal with it.

Thank you so much!

**Comment by atrolex** [comment, marked as answer]
Hi Marcus,

we have a paid module "advanced data types". With the module you can have additional data types, which are more complex, like those which comes directly out of the box.

Data type "Composition" (currently still on the roadmap) will enable to define attribute, which values will consists of values for other attributes.

So you can have 3 attributes:
1) Min Temp = 10C
2) Max Temp = 20C and
3) Temp = 10..20C or 10-20C or something else
In the definition of the "Temp" attribute you define a template how its virtual value should be created, eg "{{min_temp}}..{{max_temp}}"

Without this module you can use data type "string", but you will need to edit this value each time you edit min or max temperature.

**Reply by marcus-at-localhost** [reply]
You thought of everything :) That Advanced Data Type modul sounds phantastic. I'll keep an eye on that module. Thank you!
(& Happy Holidays)
